docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,106 @@
+## CTCLoss <a name="CTCLoss"></a>
+
+**Versioned name**: *CTCLoss-4*
+
+**Category**: Sequence processing
+
+**Short description**: *CTCLoss* computes the CTC (Connectionist Temporal Classification) Loss.
+
+**Detailed description**:
+
+This operation is similar [Reference](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/ctc_loss)
+
+*CTCLoss* calculates loss between input and target sequences. It sums over the probability of possible alignments of input to target.
+
+Input sequences `inputs[:,i,:]` can have different length. The lengths of sequences are coded as values 1 and 0 in the second input tensor `sequence_mask`. 
+Value `sequence_mask[j, i]` specifies whether there is a sequence symbol at index `j` in the sequence `i` in the batch of sequences. 
+If there is no symbol at `j`-th position `sequence_mask[j, i] = 0`, and `sequence_mask[j, i] = 1` otherwise. 
+Starting from `j = 0`, `sequence_mass[j, i]` are equal to 1 up to the particular index `j = last_sequence_symbol`, 
+which is defined independently for each sequence `i`. For `j > last_sequence_symbol`, values in `sequence_mask[j, i]` are all zeros.
+A lenght of target sequence from `labels[i,:]` must not be greater than a lenght of corresponding input sequence. Otherwise, the operation behaves unexpectedly.
+
+*CTCLoss* calculation scheme:
+
+1. Compute probability of `j`-th character at time step `t` for `i`-th input sequence from `inputs` using softmax formula:
+\f[
+p_{t,i,j} = \frac{\exp(inputs[t,i,j])}{\sum^{K}_{k=0}{\exp(inputs[t,i,k])}}
+\f]
+
+2. For a given `i`-th target from `labels[i,:]` find all aligned paths.
+A path `S = (c1,c2,...,cT)` is aligned with a target `G=(g1,g2,...,gT)` if both chains are equal after decoding.
+The decoding removes pads `-1` from a target `G` and merges repeated characters in `G` in case *preprocess_collapse_repeated* equal to True.
+The decoding merges repeated characters in `S` in case *merge_repeated* equal to True and removes blank characters represented by last class index `K`, 
```

Comment:
`blank characters represented by last class index `K`` - I don't quite understand this statement. What does "last" mean? Following the examples below you refer to the last element in the sequence. But is it really the case? According to the TF specification: `largest value (num_classes - 1) is reserved for the blank label`, so I think that the examples below are wrong.

Reply:
You are correct. I need to fix it. Indeed `K` is a number of classes including a blank so `K-1` will assign to a blank in this case. And in the example below `K` = 5 so blank is coded as `4`. I will fix it.
Great catch!

Reply:
These sentences with description about the decoding should be updated since `blank` index is specified through separate input and can be different from `N-1`.

Reply:
Fixed.